src/AutoMapper.Extensions.ExpressionMapping/XpressionMapperVisitor.cs

```diff
@@ -80,7 +75,10 @@ protected override Expression VisitMember(MemberExpression node)
                 });
 
                 fullName = BuildFullName(beforeCustExpression);
-                var visitor = new PrependParentNameVisitor(last.CustomExpression.Parameters[0].Type/*Parent type of current property*/, fullName, InfoDictionary[parameterExpression].NewParameter);
+                var visitor = new PrependParentNameVisitor(
+                    last.CustomExpression.Parameters[0].Type/*Parent type of current property*/, 
+                    fullName, 
+                    InfoDictionary[parameterExpression].NewParameter);
```

Comment:
Please replace `InfoDictionary[parameterExpression].NewParameter` with `visitedBaseExpr`.  This will fail for custom expressions e.g. `cfg.CreateMap<Entity, DTO>()
                        .ForMember(dest => dest.Description, opts => opts.MapFrom(src => string.Concat(src.Id.ToString(), " - ", src.Name)))`

Reply:
Done.